pkg/protocol/engine/engine.go

```diff
@@ -183,7 +183,7 @@ func New(
 				}
 
 				// Only mark any pruning indexes if we loaded a non-genesis snapshot
-				if e.Storage.Settings().LatestFinalizedSlot() > 0 {
+				if e.Storage.Settings().LatestFinalizedSlot() > e.Storage.GenesisRootBlockID().Slot() {
```

Comment:
Don't we have protoParams.GenesisSlot available here?

Reply:
Yes, I used this one because it was shorter than `e.Storage.Settings().APIProvider().CommittedAPI().ProtocolParameters().GenesisSlot()`, but I can change it if you like that better.